source/ht_component_factory.cpp

```diff
@@ -0,0 +1,119 @@
+
+#include "ht_component_factory.h"
+
+/*[[[cog
```

Comment:
I have no idea what to do about this.  @Honeybunch Any ideas, or do we leave it like this?